src/web/page/web_page.rs

```diff
@@ -1,14 +1,20 @@
 use super::TemplateData;
 use iron::{headers::ContentType, response::Response, status::Status, IronResult, Request};
 use serde::Serialize;
+use std::borrow::Cow;
 use tera::Context;
 
 /// When making using a custom status, use a closure that coerces to a `fn(&Self) -> Status`
 #[macro_export]
 macro_rules! impl_webpage {
+    ($page:ty = $template:literal $(, status = $status:expr)? $(, content_type = $content_type:expr)? $(,)?) => {
+        impl_webpage!($page = |_| ::std::borrow::Cow::Borrowed($template) $(, status = $status)? $(, content_type = $content_type)?);
+    };
     ($page:ty = $template:expr $(, status = $status:expr)? $(, content_type = $content_type:expr)? $(,)?) => {
         impl $crate::web::page::WebPage for $page {
-            const TEMPLATE: &'static str = $template;
+            fn template(&self) -> ::std::borrow::Cow<'static, str> {
+                $template(self)
+            }
```

Comment:
It's better to constrain the type of the expression for better errors like we do in `get_status`
```suggestion
            fn template(&self) -> ::std::borrow::Cow<'static, str> {
                let template: fn(&Self) -> ::std::borrow::Cow<'static, str> = $template;
                template(self)
            }
```